pkg/controller/miganalytic/miganalytics_controller.go

```diff
@@ -0,0 +1,589 @@
+
+		err = r.Update(context.TODO(), analytic)
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+		log.Info("Completed analysis of namespace.", "namespace", namespace)
+	}
+	log.Info("Completed analysis of plan.", "plan", plan.Name)
+	return nil
+}
+
+func (r *ReconcileMigAnalytic) analyzeK8SResources(dynamic dynamic.Interface, resources []*metav1.APIResourceList, namespace *migapi.MigAnalyticNamespace, excludedResources []string, incompatible v1alpha1.Incompatible) error {
+	gvk.SortResources(resources)
+	cohabitatingResources := gvk.NewCohabitatingResources()
+
+	for _, res := range resources {
+		for _, r := range res.APIResources {
+			// skip resource if we have already handle it via an alternate group
+			// i.e. apps/v1beta1 deployment vs extension/v1beta1 deployment
+			if cohabitator, found := cohabitatingResources[r.Name]; found {
+				if cohabitator.Seen {
+					continue
+				}
+				cohabitator.Seen = true
+			}
+
+			excluded := false
+			compatible := true
+			gvr := schema.GroupVersionResource{}
+			gv := strings.Split(res.GroupVersion, "/")
+			if len(gv) > 1 {
+				gvr = schema.GroupVersionResource{
+					Group:    gv[0],
+					Version:  gv[1],
+					Resource: r.Name,
+				}
+			} else {
+				gvr = schema.GroupVersionResource{
+					Group:    "",
+					Version:  gv[0],
+					Resource: r.Name,
+				}
+			}
+
+			list, err := dynamic.Resource(gvr).Namespace(namespace.Namespace).List(metav1.ListOptions{})
+			if err != nil {
+				return liberr.Wrap(err)
+			}
+
+			// If no resources of this type we won't add it to any lists
+			if len(list.Items) > 0 {
+
+				// Check if the resource type is on the plan exludedResources list
+				excluded = isExcluded(gvr.Resource, excludedResources)
+
+				// The lists are mutually exclusive. Only if not excluded check the incompatible GVK list
+				if !excluded {
+					compatible = isCompatible(gvr, namespace.Namespace, incompatible.Namespaces)
+				}
+
+				NamespaceResource := migapi.MigAnalyticNamespaceResource{
+					Group:   gvr.Group,
+					Version: gvr.Version,
+					Kind:    gvr.Resource,
+					Count:   len(list.Items),
+				}
+				if !compatible {
+					namespace.IncompatibleK8SResources = append(namespace.IncompatibleK8SResources, NamespaceResource)
+					namespace.IncompatibleK8SResourceTotal += len(list.Items)
+				} else if excluded {
+					namespace.ExcludedK8SResources = append(namespace.ExcludedK8SResources, NamespaceResource)
+					namespace.ExcludedK8SResourceTotal += len(list.Items)
+				} else {
+					namespace.K8SResources = append(namespace.K8SResources, NamespaceResource)
+					namespace.K8SResourceTotal += len(list.Items)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (r *ReconcileMigAnalytic) analyzeImages(client k8sclient.Client, namespace *migapi.MigAnalyticNamespace, listImages bool, listImagesLimit int) error {
+	imageStreamList := imagev1.ImageStreamList{}
+	internalRegistry, err := GetRegistryInfo(4, 4, client)
```

Comment:
Why is `(4, 4, client)` being passed in here? It seems like `GetRegistryInfo()` is parsing a k8s major/minor version but this looks like an OpenShift version. Why are we hard-coding 4,4?

Reply:
It shouldn't be hard coded. I need to fix this. The code was copied from openshift-velero-plugin. I think I was missing a GetServerVersion function and hard coded it. I honestly don't even remember it. I'll fix it.

https://github.com/konveyor/openshift-velero-plugin/blob/76906dcc99cf3f5007c7d074461b822b86a580e5/velero-plugins/common/shared.go#L88-L122

Reply:
Should be fixed. Tested quickly and seems to still be working as expected.